### PR TITLE
NOTIF-150 Introduce behavior groups - Phase 2

### PR DESCRIPTION
**Pull request by gwenneg**
This is phase 2 of #170.

You shall not merge

We're not ready 😄

See the migration plan (Google doc) for more details about this.

### CODE REVIEW

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java [hunk 1]
```diff
@@ -131,47 +128,19 @@ private String getAddress(Principal principal) {
...
     /*
      * Called by the UI to build the endpoint removal confirmation screen.
      * That screen shows all the event types (and their application) that will be affected by the endpoint removal.
      */
     @GET
-    @Path("/bg/eventTypes/affectedByRemovalOfEndpoint/{endpointId}") // TODO [BG Phase 2] Remove '/bg' path prefix
+    @Path("/eventTypes/affectedByRemovalOfEndpoint/{endpointId}")
     @Produces(APPLICATION_JSON)
     @Operation(summary = "Retrieve the event types affected by the removal of an integration.")
     @Tag(name = OApiService.PRIVATE)
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:139]: Could you remove these too?

Comment (gwenneg) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:139]: Do you mean that `GET /eventTypes/affectedByRemovalOfEndpoint/{endpointId}` is no longer needed after the behavior group migration?

**Review by josejulio — APPROVED** [review]
lgtm

File: src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java [hunk 2]
```diff
@@ -131,47 +127,18 @@ private String getAddress(Principal principal) {
...
-    @Path("/bg/eventTypes/affectedByRemovalOfEndpoint/{endpointId}") // TODO [BG Phase 2] Remove '/bg' path prefix
+    @Path("/eventTypes/affectedByRemovalOfEndpoint/{endpointId}")
     @Produces(APPLICATION_JSON)
     @Operation(summary = "Retrieve the event types affected by the removal of an integration.")
-    @Tag(name = OApiService.PRIVATE)
     @RolesAllowed(RbacIdentityProvider.RBAC_READ_NOTIFICATIONS)
     public Uni<List<EventType>> getEventTypesAffectedByRemovalOfEndpoint(@Context SecurityContext sec, @PathParam("endpointId") UUID endpointId) {
         return getAccountId(sec)
-                .onItem().transformToUni(accountId -> apps.getEventTypesByEndpointId_BG(accountId, endpointId));
+                .onItem().transformToUni(accountId -> apps.getEventTypesByEndpointId(accountId, endpointId));
```
Comment (josejulio) [inline src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:141]: I missed this before, but I think we would like to get the behavior groups affected by the removal of the endpoint id.
I'll check with UX.

Comment (gwenneg) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:141]: In this PR we have:
- `GET /eventTypes/affectedByRemovalOfEndpoint/{endpointId}` [1]
- `GET /eventTypes/affectedByRemovalOfBehaviorGroup/{behaviorGroupId}` [2]

Are you saying we will remove [1] and [2] and also introduce `GET /behaviorGroups/affectedByRemovalOfEndpoint/{endpointId}`? These APIs are so similar that I want to make sure I understood correctly 😃

Comment (josejulio) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java:141]: We will remove [1].
We want to know what behavior groups are affected by the removal of an endpoint.

We also need [2], so we don't need to touch it :).

Comment (gwenneg) [reply src/main/java/com/redhat/cloud/notifications/routers/NotificationService.java]: I pushed a commit with the `affectByRemoval*` APIs changes.

The new API returns `BehaviorGroup` including their actions, but I'm not sure actions are needed there. Please let me know if I should remove the actions from the REST response.

**Review by josejulio — APPROVED** [review]

**Comment by gwenneg** [comment]
Squashed commits, no code change.

**Pull request merged**